src/software/redis-server/redis_test.cpp

```diff
@@ -0,0 +1,38 @@
+#include <iostream>
+#include <string>
+
+#include "cpp_redis/core/client.hpp"
+#include "software/test_util/test_util.h"
+
+TEST(ServerTest, Get)
+{
+    cpp_redis::client client;
+    client.connect(
+        "127.0.0.1", 6379,
+        [](const std::string &host, std::size_t port, cpp_redis::connect_state status) {
+            if (status == cpp_redis::connect_state::dropped)
+            {
+                std::cout << "client disconnected from " << host << ":" << port
+                          << std::endl;
+            }
+        });
+
+    client.set("A", "1", [](cpp_redis::reply &reply) {
+        std::cout << "Setting A to 1: " << reply << std::endl;
+    });
+
+    client.get("A", [](cpp_redis::reply &reply) {
+        std::cout << "Getting A: " << reply << std::endl;
+        ASSERT_EQ(atoi(reply.as_string().c_str()), 1);
+    });
+
+    client.set("A", "2", [](cpp_redis::reply &reply) {
+        std::cout << "Setting A to 2: " << reply << std::endl;
+    });
+
+    client.get("A", [](cpp_redis::reply &reply) {
+        std::cout << "Getting A: " << reply << std::endl;
+        ASSERT_EQ(atoi(reply.as_string().c_str()), 2);
+    });
+    client.sync_commit();
```

Comment:
When you did redis speed tests, were you calling sync_commit every time? 

I.e can we call client.get("robot_id"); client.sync_commit(); at 200hz and not bog things down?

Reply:
When I ran the redis speed tests, it was actually using the cli so it was a lot faster. From my previous testing, it seems like a request through the CLI takes approximately ~0.12ms to complete, I added an additional speed test in this test file that tests 200 requests and logs the elapsed time and it approximately takes 0.25s to finish which means it takes ~1.25ms to complete each request. Although it's 10x slower than the CLI version, I don't think this would cause a significant latency issue, what are your thoughts?

Reply:
I think that should be good enough

Do you know if its possible to trigger a callback on change? (nonblockign for this PR though)

Reply:
What do you mean by a callback on change? Like a redis-cpp function that gets called after storing/getting data from the redis server?

Reply:
No like not polling the server, but rather being notified of an update

Reply:
If you want to be notified of an update on the redis-server we could potentially use the publish/subscribe mechanism to get notified when a key has been changed?

Reply:
ah gotcha, we can look into that in a later PR